doc/interfaces/registration_flow.rst

```diff
@@ -0,0 +1,116 @@
+.. _registration_flow:
+
+============
+Registration
+============
+
+This tutorial walks through the steps to perform image-based and streamline-based
+registration using DIPY.
+Multiple registration methods are available in DIPY.
+
+You can try these methods using your own data; we will be using the data in DIPY.
+You can check how to :ref:`fetch the DIPY data<data_fetch>`.
+
+-------------------
+Affine Registration
+-------------------
+
+Affine registration is an optimization strategy used to maximize the mutual 
+information between two volumes or images. The affine transformation allows 
+linear coordinate changes that include shearing, scaling, rotation and translation.
+
+The workflow for the affine registration requires the paths to the static image file,
+and to the moving image file. Static image is used as a reference and moving image is
+transformed during optimization.
+
+You may want to create an output directory to save the transformed image::
+
+    mkdir affine_reg_output
+
+Run the following command::
+
+    dipy_align_affine <path_to_static_file> <path_to_moving_file> --out_dir "affine_reg_output" --out_affine "affine_reg.txt"
+
+This command will apply affine transformation on the moving image file and save the
+transformed image and the affine matrix to the directory ``affine_reg_output``.
+
+In case you did not specify the output directory, the transformed image file and 
+affine matrix would be saved in the input directory by default. If you did not
+specify the name of the output affine matrix, the affine matrix will be saved to a
+file named ``affine.txt`` by default, located in the output directory also by
+default.
+
+------------------------------------
+Symmetric Diffeomorphic Registration
+------------------------------------
```

Comment:
Please add the citation/reference for this method.

Reply:
Addressed in adcad46